docs: flesh out tips for typed linting and .eslintrc.cjs (#6919)
typescript-eslint/typescript-eslint: docs/linting/Troubleshooting.mdx

Author: @jutanium would love your input on this if you have time 😇

## I get errors telling me "ESLint was configured to run ... However, that TSConfig does not / none of those TSConfigs include this file"

> These errors come are caused by an ESLint config requesting type information be generated for a file that isn't included in the TypeScript configuration.

Reviewer: ```suggestion
These errors are caused by an ESLint config requesting type information be generated for a file that isn't included in the TypeScript configuration.
```

### Fixing the Error

> <!-- prettier-ignore-start -->

Author: Am I doing this wrong? Try removing these disables - is this a bug in Prettier / its mdx parsing? Or is it just me?

- If you **do not** want to lint the file:
  - Use [one of the options ESLint offers](https://eslint.org/docs/latest/user-guide/configuring/ignoring-code) to ignore files, namely a `.eslintignore` file, or `ignorePatterns` config.
- If you **do** want to lint the file:
  - If you **do not** want to lint the file with [type-aware linting](./Typed_Linting.mdx):
    - Use [ESLint's `overrides` configuration](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns) to configure the file to not be parsed with type information:
      <details>
        <summary>

>           A popular setup is remove all typed rules from top-level configuration

Reviewer: ```suggestion
          A popular setup is to remove all rules requiring type information from the top-level configuration
```

          and only apply them to TypeScript files via an override.
        </summary>

        ```js title=".eslintrc.cjs"
        module.exports = {
          // ... the rest of your config ...
          overrides: [
            {
              extends: [
                'plugin:@typescript-eslint/recommended-requiring-type-checking',
              ],
              files: ['./**/*.{ts,tsx}'],
            },
          ],
        };
        ```

      </details>

>       <details>

Reviewer 2: Maybe out this first so that the simpler config option is front and Center?

Author: Hmm interesting! Maybe once v6 lands we can? I personally lean towards side of putting additive changes in `overrides`. So maybe soon 😄

        <summary>

>           Alternatively, in our <a href="/blog/announcing-typescript-eslint-v6-beta">version v6</a>, you can add use our{' '}

Reviewer: ```suggestion
          Alternatively, in our <a href="/blog/announcing-typescript-eslint-v6-beta">version v6</a>, you can use our{' '}
```
